modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c

```diff
@@ -63,53 +75,34 @@ void nrf_802154_clock_hfclk_stop(void)
 	hfclk_is_running = false;
 }
 
-bool nrf_802154_clock_hfclk_is_running(void)
-{
-	return hfclk_is_running;
-}
+#elif defined(NRF54H_SERIES)
 
-static void lfclk_on_callback(struct onoff_manager *mgr,
-			      struct onoff_client  *cli,
-			      uint32_t state,
-			      int res)
-{
-	lfclk_is_running = true;
-	nrf_802154_clock_lfclk_ready();
-}
+#define NRF_LRCCONF_RADIO_PD NRF_LRCCONF010
 
-void nrf_802154_clock_lfclk_start(void)
+static void hfclk_started_timer_handler(struct k_timer *dummy)
 {
-	int ret;
-	struct onoff_manager *mgr =
-		z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_LF);
-
-	__ASSERT_NO_MSG(mgr != NULL);
-
-	sys_notify_init_callback(&lfclk_cli.notify, lfclk_on_callback);
-
-	ret = onoff_request(mgr, &lfclk_cli);
-	__ASSERT_NO_MSG(ret >= 0);
+	hfclk_is_running = true;
+	nrf_802154_clock_hfclk_ready();
 }
 
-void nrf_802154_clock_lfclk_stop(void)
-{
-	int ret;
-	struct onoff_manager *mgr =
-		z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_LF);
+K_TIMER_DEFINE(hfclk_started_timer, hfclk_started_timer_handler, NULL);
 
-	__ASSERT_NO_MSG(mgr != NULL);
+void nrf_802154_clock_hfclk_start(void)
+{
+	/* Use register directly, there is no support for that task in nrf_lrcconf_task_trigger */
+	NRF_LRCCONF_RADIO_PD->EVENTS_HFXOSTARTED = 0x0;
+	NRF_LRCCONF_RADIO_PD->TASKS_REQHFXO = 0x1;
 
-	ret = onoff_cancel_or_release(mgr, &lfclk_cli);
-	__ASSERT_NO_MSG(ret >= 0);
-	lfclk_is_running = false;
+	k_timer_start(&hfclk_started_timer, K_USEC(1000), K_NO_WAIT);
```

Comment:
I would replace the magic number with a define, something like `#define MAX_HFXO_RAMP_UP_TIME_US 1000`. I think you can read this value from BICR instead of hardcoding here, but initially, it is ok to keep it as it is.

Reply:
Done